src/test/java/org/example/springbootintro/book/BookSpecificationBuilderTest.java

```diff
@@ -0,0 +1,91 @@
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+
+import org.example.springbootintro.dto.book.BookSearchParametersDto;
+import org.example.springbootintro.model.Book;
+import org.example.springbootintro.repository.SpecificationProvider;
+import org.example.springbootintro.repository.book.BookSpecificationBuilder;
+import org.example.springbootintro.repository.book.BookSpecificationProviderManager;
+import org.example.springbootintro.repository.book.spec.AuthorSpecificationProvider;
+import org.example.springbootintro.repository.book.spec.TitleSpecificationProvider;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.jpa.domain.Specification;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayName("Book Specification Builder Tests")
+class BookSpecificationBuilderTest {
+    @InjectMocks
+    private BookSpecificationBuilder bookSpecificationBuilder;
+
+    @Mock
+    private BookSpecificationProviderManager bookSpecificationProviderManager;
+
+    @Mock
+    private SpecificationProvider<Book> titleSpecificationProvider;
+
+    @Mock
+    private SpecificationProvider<Book> authorSpecificationProvider;
+
+    @Test
+    @DisplayName("Build with empty search parameters returns null")
+    void build_WithEmptySearchParameters_ReturnsNull() {
+        // Given
+        BookSearchParametersDto searchParametersDto = new BookSearchParametersDto(null, null);
+
+        // When
+        Specification<Book> spec = bookSpecificationBuilder.build(searchParametersDto);
+
+        // Then
+        assertNull(spec);
+    }
+
+    @Test
+    @DisplayName("Build with title search parameter returns title specification")
+    void build_WithTitleSearchParameter_ReturnsTitleSpecification() {
+        // Given
+        BookSearchParametersDto searchParametersDto =
+                new BookSearchParametersDto(new String[]{"Title"}, null);
+        when(bookSpecificationProviderManager
+                .getSpecificationProvider(TitleSpecificationProvider.TITLE))
+                .thenReturn(titleSpecificationProvider);
+        when(titleSpecificationProvider.getSpecification(any()))
+                .thenReturn((root, query, criteriaBuilder)
+                        -> criteriaBuilder.equal(root.get("title"), "Some Title"));
+
+        // When
+        Specification<Book> spec = bookSpecificationBuilder.build(searchParametersDto);
+
+        // Then
+        assertNotNull(spec);
+    }
+
+    @Test
+    @DisplayName("Build with author search parameter returns author specification")
+    void build_WithAuthorSearchParameter_ReturnsAuthorSpecification() {
+        // Given
+        BookSearchParametersDto searchParametersDto =
+                new BookSearchParametersDto(null, new String[]{"Author"});
```

Comment:
it's better to not pass nulls to the constructor, if that situation happens - use setters or add another constructor that fits the data that you need it to have. If you have too many constructors, implement a builder pattern for that class or use lombok's `@Accesors(chain = true)`